packages/entity-server/src/index.ts

```diff
@@ -3,32 +3,36 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
-import '@babel/polyfill';
-
-import * as dotenv from 'dotenv';
-
-import http from 'http';
-
-import { createApp } from './app';
-import winston from './log';
-
-dotenv.config(); // Load the environment variables into process.env
-
-/**
- * Set up app with routes etc.
- */
-const app = createApp();
-
-/**
- * Start the server
- */
-const port = process.env.PORT || 8050;
-http.createServer(app).listen(port);
-winston.info(`Running on port ${port}`);
-
-/**
- * Notify PM2 that we are ready
- * */
-if (process.send) {
-  process.send('ready');
-}
+import { PartialOrArray } from '@tupaia/server-boilerplate';
+
+import { EntityFields } from './models';
+import {
+  FlattableEntityFieldName,
+  ExtendedEntityFieldName,
+  EntityResponseObject,
+} from './routes/hierarchy/types';
+
+export type EntityFilterObject = PartialOrArray<EntityFields>;
+
+export type EntityApiQueryOptions = {
+  field?: FlattableEntityFieldName;
+  fields?: ExtendedEntityFieldName[];
+  filter?: EntityFilterObject;
+};
+export type RelationshipsSubQueryOptions = Omit<EntityApiQueryOptions, 'fields'>;
+
+export type EntityApiResponse<T extends ExtendedEntityFieldName[]> = Required<
+  Pick<EntityResponseObject, T[number]>
+>;
+
+export {
+  GroupByAncestorRelationshipsResponseBody,
+  GroupByDescendantRelationshipsResponseBody,
+} from './routes/hierarchy/relationships';
+
+export {
+  FlattableEntityFieldName,
+  ExtendedEntityFieldName,
+  EntityResponseObject,
+  FlattenedEntity,
+} from './routes/hierarchy/types';
```

Comment:
Bit sad to have to export so many internal types from the entity-server, but overall it seemed the cleanest approach. Defining the EntityApi within the entity-server was just not cutting it and re-defining these types would be ugly as well...